numba/np/arrayobj.py

```diff
@@ -3929,15 +3928,19 @@ def linspace(start, stop):
 def numpy_linspace_3(context, builder, sig, args):
     dtype = as_dtype(sig.return_type.dtype)
 
+    # Implementation based on https://github.com/numpy/numpy/blob/v1.20.0/numpy/core/function_base.py#L24 # noqa: E501
     def linspace(start, stop, num):
         arr = np.empty(num, dtype)
         if num == 0:
             return arr
         div = num - 1
-        delta = stop - start
-        arr[0] = start
-        for i in range(1, num):
-            arr[i] = start + delta * (i / div)
+        if div > 0:
+            delta = stop - start
+            step = np.divide(delta, div)
```

Comment:
Is `np.divide` used to make sure numpy error model here?

Reply:
Yes, but it's probably not necessary as the predicate would likely propagate and the div-by-zero branch would be DCE. Am happy to change.

Reply:
i'd prefer the using `/`

Reply:
Done in 947c744